veps/sig-network/link-state-proposal.md

```diff
@@ -0,0 +1,232 @@
+    ...
+}
+```
+
+Type `InterfaceState` currently has a single allowed value:  
+```go
+type InterfaceState string
+
+const (
+	InterfaceStateAbsent InterfaceState = "absent"
+)
+```
+
+The following values will be added to specify the required link state:
+```go
+const (
+    InterfaceStateLinkUp   InterfaceState = "up"
+    InterfaceStateLinkDown InterfaceState = "down"
+)
+```
+
+An empty value will be considered as `up`.
+
+The network validator will be adjusted to allow the two new values for all interfaces regardless of whether they are 
+connected to primary or secondary networks, or whether they use a core binding or a binding plugin.
+
+> [!IMPORTANT]
+> Controlling the link state of SR-IOV NICs will not be supported as we cannot control their link state via the domain.
+> 
+> Setting the desired link state of an SR-IOV NIC will not be permitted (enforced by the validating webhooks), as it is not currently possible setting it in the domain. 
+
+virt-launcher's `Converter` component will be adjusted to take the interface State field into account when creating a new domain XML.
+In case the field's value is `InterfaceStateLinkDown`, the interface will be created as follows:
+```xml
+<interface>
+  ...
+  <link state='down'/>
+  ...
+</interface>
+```
+
+> [!NOTE]
+> For additional details please see libvirt's [documentation](https://libvirt.org/formatdomain.html#modifying-virtual-link-state).
+
+virt-launcher's `LibvirtDomainManager.SyncVMI` will be extended to support updating the interfaces link state while the VM is running.
+
+The new logic will:
+1. Fetch the domain.
+2. For each interface in the VMI spec:
+ - Compare the actual link state against the desired link state.
+ - Adjust the interface entry in the domain in case there is a need to take action.
+
+virt-launcher's NIC hotplug logic will be adjusted to support hot-plugging a NIC while its link state is down.
+
+### Network Binding Plugins Support
+
+When using the functionality purposed by this enhancement with network binding plugins using a sidecar container -
+it is up to the plugin's vendor to support this functionality - as they control the interface section in the domain XML.
+
+### Link State Reporting
+
+A new field will be added to the VMI interface status that will be used for reporting the interface's current link state (up/down):
+```go
+type VirtualMachineInstanceNetworkInterface struct {
+	...
+	LinkState string `json:"linkState,omitempty"`
+	...
+}
+```
+
+virt-handler will be responsible for reporting this field's value, based on domain spec information received from virt-launcher.
+An empty specification (as exists today) would be considered as `up`.
+
+SR-IOV NICs' link state will not be reported, as it could change without KubeVirt awareness.
+
+> [!NOTE]
+> The linkState information will be reported for interfaces which are present in the VirtualMachineInstance's spec.
+> Interfaces defined inside the guest will not have their link state reported.
```

Comment:
What about the default interface when none is defined in the spec, will its state be reported?

Reply:
The default interface is automatically added to the VMI spec (unless the use had explicitly specified against it).

https://github.com/kubevirt/kubevirt/blob/a7c2595a54149662e564d89fd377f60b210570a2/pkg/network/vmispec/defaults.go#L33

Reply:
So it will be reported.